src/common/components/Input/index.tsx

```diff
@@ -1,19 +1,62 @@
 /* eslint-disable indent */
-import { createContext, useContext } from 'react';
+import { createContext, useContext, useEffect, useRef, useState } from 'react';
 
-import { circle, title, inputLine, containerVar, inputVar, descriptionVar } from './style.css';
-import { TextBoxProps } from './types';
+import Button from '@components/Button';
+import formatTimer from '@utils/formatTimer';
+
+import { circle, title, inputLine, containerVar, inputVar, descriptionVar, timer } from './style.css';
+import { InputButtonProps, TextBoxProps, TimerProps } from './types';
 
 const FormContext = createContext({} as Pick<TextBoxProps, 'required' | 'formObject'>);
 
+// TextBox 내부 InputLine 우측 버튼
+export const InputButton = ({ text, ...props }: InputButtonProps) => {
+  return (
+    <Button style={{ width: '148px' }} {...props}>
+      {text}
+    </Button>
+  );
+};
+
+// TextBox 내부 타이머
+export const Timer = ({ isActive, setActive }: TimerProps) => {
+  const [seconds, setSeconds] = useState(300);
+  const timerRef = useRef<NodeJS.Timeout | null>(null);
+
+  useEffect(() => {
+    if (isActive) {
+      setSeconds(300);
+      timerRef.current = setInterval(() => {
+        setSeconds((prev) => prev - 1);
+      }, 1000);
+    } else {
+      timerRef.current && clearInterval(timerRef.current);
```

Comment:
이 부분이 3번 반복되고 있어요! 따로 함수 빼주면 좋을 거 같습니다~

```tsx
const clearTimer = () => {
  if (timerRef.current) {
    clearInterval(timerRef.current);
    timerRef.current = null;
  }
};
```
```suggestion
    } else {
      clearTimer();
```